### DISCUSSION TITLE
Input ui.file to Excel?

### DISCUSSION THREAD

**Discussion by PJaekel93** [marimo-team/marimo#1942] [Q&A]
Hi, 
when reading a an Excel (XLSX) File via ui.file the contents return Bytes.

The pd.read_excel() function does not work with the marimo file object.

Am I missing something, is there a way to upload an excel file and convert it to a readable object, e.g. Pandas DataFrame?

Code:
`f = mo.ui.file()

f.contents()`

Output:
[Output.txt](https://github.com/user-attachments/files/16469339/Output.txt)

File used:
[Testfile.xlsx](https://github.com/user-attachments/files/16469268/Testfile.xlsx)

**Comment by akshayka** [comment]
It works on my machine.

If it's still not working for you, please provide more details. You can file a bug report on GitHub issues, which will tell you which details to include.

![image](https://github.com/user-attachments/assets/486b2f00-b2d6-4d92-b69f-e489ba62fa69)

**Reply by PJaekel93** [reply]
Many thanks, this indeed works. However, returning the contents as bytes from the uploaded file and converting to dataframe via read_excel() loses the formatting do the original input files.

**Reply by akshayka** [reply]
If you use read_excel directly (with a file path), instead of going through ui.file, is formatting preserved? I am not sure if the loss of formatting is a marimo problem.
